### PR TITLE
Fix yaml load error for ruby 3.1

### PR DESCRIPTION
**Pull request by Stromweld**
## Description
Ruby 3.1 changed the way YAML load works. See issue #218 for more details

## Related Issue
#218 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -13,43 +13,34 @@ jobs:
...
+      - uses: actions/checkout@v3
+      - name: Set up ruby 3.1
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 2.7
+          ruby-version: 3.1
           bundler-cache: true
       - name: run specs
         run: bundle exec rake spec --trace
-      - name: Simplecov Report
```
Comment (vkarve-chef) [inline .github/workflows/lint.yaml:50]: shall we restore the simplecov report run? any issue it appears to cause?

**Review by vkarve-chef — APPROVED** [review]

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (Stromweld) [reply .github/workflows/lint.yaml:50]: It keeps failing causing the whole job to fail even though it passed the rspec tests. I also checked out other chef org repos and didn't see any similar jobs so removed it.

**Comment by Stromweld** [comment]
I don't have access to merge so if you could merge this when you get the chance that'd be great.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (vkarve-chef) [reply .github/workflows/lint.yaml]: i recalled this step fails for PRs from fork, so restored it

**Pull request merged**